### ISSUE TITLE
studiomdl.exe improvements for better models

### ISSUE THREAD

**Issue by AutismoPiggo** [Facepunch/garrysmod-requests#1526]
For consideration, would make porters work a lot easier and better quality 
-support all DMX binary versions, like sfm
-add dmxeyelid support
-remove the material limit on models (its 32)
-remove the flex vert limit on models (its 10 000, also will require to change the code in gmod since it just crashes when trying to facepose with models over the limit)
-remove the vertices limit (its somewhere around 70 000)
-remove the bone limit (or at least increase it to 256/512)
-maybe increase number of weights per vertex

**Comment by Kinyom** [comment]
One thing that could also be useful is being allowed to use quaternions as angles in .smds. It really wouldn’t be much work to add (especially considering that studiomdl already converts Euler rotations to quaternions iirc, and .dmx files already use them), and could be done without impacting reading of current .smds at all. Hell, if that’s a concern even then a special flag that needs to be set at the top of the file could even work.

**Comment by Roachdaripper** [comment]
removing the bone limit would be a godsend, considering every modder seems to be competing with one another on who can put models from the latest games into gmod, and more often than not a lot of sacrifices have to be made to get these things under the bone limit.

**Comment by One-Trick-Viper** [comment]
yep, these changes would definitely help. especially removing the bone limit would help because a lot of newer games tend to go over this bone limit.

**Comment by TroubleAlwaysInbound** [comment]
This would benefit the community so much by increasing the quality of add-ons available. So many mods are held back by the current limitations, which is quite unfortunate. +1 for this

**Comment by OlmateUbafest** [comment]
100% this. These limits feel arbitrary as hell and serve no real purpose when the engine can very clearly handle it

Characters with facial hair for example are super problematic, but when you separate elements and then merge them together in-game you get no weird behavior or crashes

**Comment by TheFrostBible** [comment]
Yeah, this would be really help anyone doing model work. The material/vert/bone limits do nothing but obstruct many of the people that try to work with GMod's studio.mdl.

**Comment by Coinages** [comment]
This would 100% make a modder's life significantly easier, especially increasing the poly count

**Comment by AutismoPiggo** [comment]
Updated this request to make the request actually understandable

Removed flex vert limit request to make new issue for seperate issue

**Comment by robotboy655** [comment]
Bone limit is already 256 since before this was posted. Going higher may be more problematic.

Do you have a model example of "dmxeyelid"? Sources and a compiled model would be needed, so I can test gmod's studiomdl versus known good model.

**Comment by KalinKD339** [comment]
Is it possible to make the compiler support multiple skingroups? or upgrade to 64bit version?
